backend/src/main/java/gov/cdc/usds/simplereport/api/DefaultArgumentValidation.java

```diff
@@ -0,0 +1,112 @@
+package gov.cdc.usds.simplereport.api;
+
+import graphql.GraphQLError;
+import graphql.GraphqlErrorBuilder;
+import graphql.execution.ResultPath;
+import graphql.language.SourceLocation;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLDirective;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLFieldsContainer;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLList;
+import graphql.schema.GraphQLModifiedType;
+import graphql.schema.GraphQLScalarType;
+import graphql.schema.GraphQLType;
+import graphql.validation.rules.ValidationEnvironment;
+import graphql.validation.rules.ValidationRule;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class DefaultArgumentValidation implements ValidationRule {
+  private static final Set<String> DIRECTIVES_OVERRIDING_DEFAULT_RULE = Set.of("Size", "Pattern");
+  private static final Set<String> SCALARS_APPLIED_TO =
+      Set.of("ID", "String", "LocalDate", "DateTime");
+  private final int maxLength;
+
+  public DefaultArgumentValidation(int maxLength) {
+    this.maxLength = maxLength;
+  }
+
+  @Override
+  public boolean appliesTo(
+      GraphQLFieldDefinition fieldDefinition, GraphQLFieldsContainer fieldsContainer) {
+    return false;
+  }
+
+  @Override
+  public boolean appliesTo(
+      GraphQLArgument argument,
+      GraphQLFieldDefinition fieldDefinition,
```

Comment:
What do these other arguments do, in theory? (I recognize that you're constrained by the interface, just wondering why the interface would include them.)

Reply:
`argument` is the argument directly targeted by the validation directive, `fieldDefinition` is the method housing the argument, and `fieldsContainer` is the object of which the method is a property.

So for 

```graphql
type Query {
  patient(
    id: ID! @Size(min: 32, max: 32)
  ): Patient
  ...
}
```

`argument` would be the argument definition for `id`, `fieldDefinition` would be the definition for `patient`, and `fieldsContainer` would be the definition for `Query`.